### DISCUSSION TITLE
Welcome to "Unnoficial Raspberry Pi Godot Engine" Discussions!

### DISCUSSION THREAD

**Discussion by hiulit** [hiulit/Unofficial-Godot-Engine-Raspberry-Pi#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tebz** [comment]
Hi.
I'm new to this, and I would like to be able to use the GPIO, or other functions on the IO-header, on Raspberry PI.
Is there a way to do this ?
Pleas help !

**Reply by hiulit** [reply]
Hey, sorry! I don't know if that's possible.

**Reply by tebz** [reply]
After sleeping on it, I think there might be two ways:
Using the command line raspi-gpio get X, raspi-gpio set X Y
Using a separate Phyton  script communication using a local UDP port.
At the moment I'm busy with other things, but I'm planing on trying it, planing, soon, maybe… 
Thanks for the reply !

**Reply by hiulit** [reply]
It would be awesome if you can make it work and share it with us :)

**Reply by tebz** [reply]
I finally taken two days to learn Godot enough to make the application.
Very experimental, but seems to work. Only for GPIO.
[RB-PI-Simple-IO.zip](https://github.com/hiulit/Unofficial-Godot-Engine-Raspberry-Pi/files/10203306/RB-PI-Simple-IO.zip)

**Comment by SysError99** [comment]
I agree with the GPIO support. Since Godot is on the way of considering official ARM64 build, this project can now focus on more Raspberry Pi specific features. Still, for sure we need more maintainers to actually add a support since this project, as for now, aims for ports only.

For the current state that Raspberry Pi is truly lacking is the support for C#, but I can't blame anything since doing C# support on RPi ARM version is still pretty hit and miss when doing things outside of server stuffs. We may be able to focus on how to port `dotnet` build of RPi or even contribute it into Godot's main repository when the actual `dotnet` version become mature enough.

**Comment by Sangeeth-S-Santhosh** [comment]
Hello, 
I am a total newbie at this , don't know why but I can't upload videos in this unofficial version of godot. Video_player_node seems not responding to any kind of video format. Isn't it possible or am I just stupid??
Sorry if it is an unnecessary discussion topic...

**Reply by SysError99** [reply]
Video player is very glitchy even on official platform versions and OGV is only one that's fully supported. If you have no luck with OGV then using video frames with audio may simulate it.

**Reply by Sangeeth-S-Santhosh** [reply]
Thanks for the reply ,
Yes I tried Ogv with theora and there was no response.Meanwhile framing video takes too much space and sometimes its more glitchier due to huge load in the background..

**Reply by Sangeeth-S-Santhosh** [reply]
Update ..
Video format does support but only in a 64 bit OS
So we need to install the "64 bit version of unofficial godot" in a "raspi with 64 bit OS"..
